process/coordinator/process.go

```diff
@@ -950,7 +973,9 @@ func (tc *transactionCoordinator) processCompleteMiniBlock(
 ) error {
 
 	snapshot := tc.accounts.JournalLen()
-	tc.initProcessedTxsResults()
+	if tc.shardCoordinator.SelfId() != core.MetachainShardId {
```

Comment:
Metachain will do this initialization upper, just before to start processing the given shard hdr, because it needs to fully process the whole shard hdr and it needs to keep track for all the scrs created after each mb processed, to be able to revert all in case of processing of one mb will fail later. In the case of shard node this initialization should be done here, just before to start processing the given mb, because it needs to fully process the whole mb instead of the whole metablock.